.github/workflows/cli-verification.yaml

```diff
@@ -0,0 +1,41 @@
+name: CLI Publish & Verify
+
+on:
+    push:
+        branches:
+            - master
+            - staging/**
+    pull_request:
+
+concurrency:
+    group: verify-cli-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    publish-and-test:
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            packages: write
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: '18.x'
+                  registry-url: 'https://npm.pkg.github.com'
+                  scope: '@nangohq'
+            - name: Publish npm packages to the github registry
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              shell: bash
+              run: |
+                  GIT_HASH=$(git rev-parse --short HEAD)
+                  bash ./scripts/publish.sh 0.0.1-$GIT_HASH true
```

Comment:
I would personally prefer the script to handle this very specific version rather than having a bypass mechanism that make it possible to misuse the script.

Reply:
same ☝🏻 but I'm also wondering why we publish and then call the script? Seems dangerous or unnecessary